lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -332,6 +332,41 @@ class PluginState(BaseStrEnum):
     WAITING_FOR_UPGRADE = "waiting-for-upgrade"
 
 
+class OpenSearchPluginRelationsHandler(ABC):
```

Comment:
at which layer will this class be extended? given we already have the plugins classes

Reply:
The plugin manager, opensearch backups, i.e. objects the charm creates. However, we may call this handler's creation several times.

Reply:
This class will act as a connection between the `plugin_manager`, which only leads with OpenSearch overall APIs; and any other plugin-specific APIs.

Reply:
The main issue here is that I need the databag at `plugin_manager` to do some configurations. However, we now have `small` and `large` deployments, which means more than one relation is possible for the same plugin.

That is becoming messy from the plugin_manager's perspective.

This class allows a given plugin (e.g. OpenSearchBackup) to instantiate a singleton that holds the databag according to its current config (e.g. using the s3-credentials or the peer-cluster relation) and the `plugin_manager` does not need to know anything else.